docs/intro/overrides.rst

```diff
@@ -58,84 +85,223 @@ Let's take a look at how the following code is structured:
-      that doesn't contain ``/digital-goods/`` in its URL path.
+    - On the other hand, ``AnotherExampleProductPage`` is used instead of
+      ``GenericProductPage`` when we're handling pages that match with the
+      ``another.example`` URL Pattern which doesn't contain ``/digital-goods/``
+      in its URL path.
 
 .. tip::
 
-    The URL patterns declared in the ``@handle_urls`` annotation can still be
+    The URL patterns declared in the ``@handle_urls`` decorator can still be
     further customized. You can read some of the specific parameters in the
     :ref:`API section <api-overrides>` of :func:`web_poet.handle_urls`.
 
+.. _item-class-example:
+
+Item Class
+~~~~~~~~~~
+
+An alternative approach for the Page Object Overrides example above is to specify
+the returned Item Class. For example, we could change the previous example into
+the following:
+
+
+.. code-block:: python
+
+    from web_poet import handle_urls, WebPage
+
+
+    class GenericProductPage(WebPage[Product]):
+        def to_item(self) -> Product:
+            return Product(product_title=self.css("title::text").get())
+
+
+    @handle_urls("some.example")
+    class ExampleProductPage(WebPage[Product]):
+        ...  # more specific parsing
+
+
+    @handle_urls("another.example", exclude="/digital-goods/")
+    class AnotherExampleProductPage(WebPage[Product]):
+        ...  # more specific parsing
+
+
+    @handle_urls(["dual.example/shop/?product=*", "uk.dual.example/store/?pid=*"])
+    class DualExampleProductPage(WebPage[Product]):
+        ...  # more specific parsing
+
+Let's break this example down:
+
+    - The URL patterns are exactly the same as with the previous code example.
+    - The ``@handle_urls`` is able to derive the returned Item Class (i.e.
+      ``Product``) from the respective Page Object. For advanced use cases, you
+      can pass the ``to_return`` parameter directly to the ``@handle_urls``
+      decorator where it replaces any derived values.
+    - The ``instead_of`` parameter can be omitted in lieu of the derived Item
+      Class from the Page Object which becomes the ``to_return`` attribute in
+      :class:`~.ApplyRule` instances. This means that:
+
+        - If a ``Product`` Item Class is requested for URLs matching with the
+          "some.example" pattern, then the ``Product`` Item Class would come from
+          the ``to_item()`` method of ``ExampleProductPage``.
+        - Similarly, if a page with a URL matches with "another.example" without
+          the "/digital-goods/" path, then the ``Product`` Item Class comes from
+          the ``AnotherExampleProductPage`` Page Object.
+        - However, if a ``Product`` Item Class is requested matching with the URL
+          pattern of "dual.example/shop/?product=*", a ``SimilarProduct``
+          Item Class is returned by the ``DualExampleProductPage``'s ``to_item()``
+          method instead.
```

Comment:
I believe it's still worth providing a standalone explanation here as it could be the case that the reader specifically traverses the docs via `Overrides -> Creating Overrides -> Item Class`.

But yeah, some of the information could be redundant if they've already read `Overrides -> Creating Overrides -> Page Object`.

I'm inclined to still leave this on as it doesn't bring much downsides apart from repeating information. But I consider that as an advantage as this subsection can stand on itself.

Reply:
> But I consider that as an advantage as this subsection can stand on itself.

I agree that this is a good goal to have, that it is good for sections to stand on their own. However, this section starts with “An alternative approach for the Page Object Overrides example above”, which I believe sets the expectation for the user to read the example above before they continue reading.

That said, I have no strong opinion here, so moving on.